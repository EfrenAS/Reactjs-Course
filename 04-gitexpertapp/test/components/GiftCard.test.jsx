import { render, screen } from '@testing-library/react'
import GiftCard from '../../src/components/GifCard'

describe('GiftCard test', () => {
  const title = 'Goku vs. Superman'
  const url = 'https://localhost:3000/goku-vs-superman.gif'

  test('should render correctly with snapshot', () => {
    const { container } = render(<GiftCard title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('should render title', () => {
    render(<GiftCard title={title} url={url} />)

    const {src, alt } = screen.getByRole('img')
    
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should render title and url', () => {
    render(<GiftCard title={title} url={url} />)

    const titleElement = screen.getByText(title)

    expect(titleElement).toBeTruthy()
  })
})