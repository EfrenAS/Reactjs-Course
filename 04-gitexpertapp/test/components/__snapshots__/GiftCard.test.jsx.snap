// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCard test should render correctly with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku vs. Superman"
      src="https://localhost:3000/goku-vs-superman.gif"
    />
    <p>
      Goku vs. Superman
    </p>
  </div>
</div>
`;
